controllers/login.js

```diff
@@ -133,13 +133,9 @@ router.all('/login/superhero/', (req, res, next) => {
 	handleLoginFailed(req, res).catch(next);
 });
 
-const ssoSchoolData = (req, systemId) => api(req).get('/schools/', {
-	qs: {
-		systems: systemId,
-	},
-}).then((schools) => {
-	if (schools.data.length > 0) {
-		return schools.data[0];
+const ssoSchoolData = (req, systemId) => api(req).get('/schoolsList').then((schools) => {
+	if (schools.length > 0) {
+		return schools.find((school) => school.systems.find((system) => system._id === systemId));
 	}
 	return undefined;
 }).catch(() => undefined); // TODO: fixme this is a very bad error catch
```

Comment:
In error case we get no information. Can we log an error and return the 2 retun undefined with empty array? 
We must only replace the line 168 if (schools.length > 0)

Reply:
I rewrote this part. Should be more simple, the code is moved to line 160. There we request from cache the list of all schools, to check if the school and systemId match with the one from result.